### PR TITLE
Sound filters v3

### PR DESCRIPTION
**Pull request by qkdefus**
Split soundfilter.json to multiple files for easyer maintaining
Added more filters from previous non committed builds

TODO : figure out how json files can read "0x0" strings since i dont feel like converting hex to int manually. 
also pasting hex into json files encapsulates the hex value with "" automatically

enjoy

### CODE REVIEW

**Comment by Reetus** [comment]
So this PR doesn't work because of hex instead of base10? if so, convert it to draft until it works

**Comment by qkdefus** [comment]
This PR work, its not that

File: ClassicAssist/ClassicAssist.csproj [hunk 1]
```diff
@@ -21,6 +28,24 @@
     <None Remove="zlib64.dll" />
   </ItemGroup>
   <ItemGroup>
+    <Content Include="Data\Filters\Audio\Player.json">
```
Comment (Reetus) [inline ClassicAssist/ClassicAssist.csproj:31]: Set to PreserveNewest

File: ClassicAssist/Data/Filters/Audio/Environment.json [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+            0xF2,
+            0xF3,
+            0xED,
+            0xF4
+        ]
+    },
+    {
+        "Category": "Environment",
+        "DefaultEnabled": false,
+        "Name": "BackPack Sounds", // qk // TODO // taken from containers.txt -> // doesnt seem to be working -> client side ?
```
Comment (Reetus) [inline ClassicAssist/Data/Filters/Audio/Environment.json:29]: If they don't work, don't include them

File: ClassicAssist/Data/Filters/Audio/Environment.json [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+            0x058,
+            0x04F,
+            0x187,
+            0x1C9
+        ]
+    },
+    {
+        "Category": "Environment",
+        "DefaultEnabled": false,
+        "Name": "FootStep Sounds", // qk // TODO // grabbed from ClassicUO Mobile.cs // doesnt seem to be working -> client side ?
```
Comment (Reetus) [inline ClassicAssist/Data/Filters/Audio/Environment.json:45]: If they don't work, don't include them

File: ClassicAssist/Data/Filters/SoundFilter.cs [hunk 4]
```diff
@@ -32,28 +32,62 @@
...
     public class SoundFilter : DynamicFilterEntry, IConfigurableFilter
     {
-        private const string DATA_FILE = "SoundFilters.json";
+        //private const string DATA_FILE = "SoundFilters.json";
 
         public SoundFilter()
         {
-            string dataPath = Path.Combine( Engine.StartupPath ?? Environment.CurrentDirectory, "Data" );
+            string dataPath = Path.Combine(Engine.StartupPath ?? Environment.CurrentDirectory, "Data\\Filters\\Audio\\");
+            //System.Windows.Forms.MessageBox.Show("" + dataPath);
```
Comment (Reetus) [inline ClassicAssist/Data/Filters/SoundFilter.cs:43]: Remove debug line

File: ClassicAssist/Data/Filters/SoundFilter.cs [hunk 5]
```diff
@@ -32,28 +32,62 @@
 
 namespace ClassicAssist.Data.Filters
 {
-    [FilterOptions( Name = "Sound Filter", DefaultEnabled = false )]
+    [FilterOptions(Name = "Sound Filter", DefaultEnabled = false)]
     public class SoundFilter : DynamicFilterEntry, IConfigurableFilter
     {
-        private const string DATA_FILE = "SoundFilters.json";
+        //private const string DATA_FILE = "SoundFilters.json";
```
Comment (Reetus) [inline ClassicAssist/Data/Filters/SoundFilter.cs:38]: Remove if Unused

File: ClassicAssist/Data/Filters/SoundFilter.cs [hunk 6]
```diff
@@ -32,28 +32,62 @@
...
+                string[] subdirectoryEntries = Directory.GetDirectories(targetDirectory);
+                foreach (string subdirectory in subdirectoryEntries)
+                {
+                    ProcessDirectory(subdirectory);
+                }
+
+            }
+            catch (Exception ex)
+            {
+                System.Windows.Forms.MessageBox.Show("" + ex);
```
Comment (Reetus) [inline ClassicAssist/Data/Filters/SoundFilter.cs:71]: Show( ex.toString() )

File: ClassicAssist/Data/Filters/SoundFilter.cs [hunk 7]
```diff
@@ -32,28 +32,62 @@
...
             Items.AddRange(
                 JsonConvert.DeserializeObject<SoundFilterEntry[]>(
-                    File.ReadAllText( Path.Combine( dataPath, DATA_FILE ) ) ) );
+                    File.ReadAllText(path)));
 
-            foreach ( SoundFilterEntry item in Items )
+            foreach (SoundFilterEntry item in Items)
             {
-                item.LocalizedName = Strings.ResourceManager.GetString( item.Name ) ?? item.Name;
-                item.Category = Strings.ResourceManager.GetString( item.Category ) ?? item.Category;
```
Comment (Reetus) [inline ClassicAssist/Data/Filters/SoundFilter.cs:56]: Include spaces between parenthesis, don't modify lines you haven't changed

**Review by Reetus — CHANGES_REQUESTED** [review]

File: ClassicAssist/Data/Filters/SoundFilter.cs (hunk 7 above)
Comment (qkdefus) [reply ClassicAssist/Data/Filters/SoundFilter.cs:56]: Its VS that does it

Comment (Reetus) [reply ClassicAssist/Data/Filters/SoundFilter.cs:56]: You need to set spaces between parenthesis in options

Comment (qkdefus) [reply ClassicAssist/Data/Filters/SoundFilter.cs:56]: Will do 👍

**Comment by Reetus** [comment]
image

Hex numbers in JSON is not valid, you will need to convert to base10

Comments is JSON also isn't supported

You've added strings without adding them to resources for localization, ie "Item Drop Sounds" etc

Fix the issues and rebase your branch and create a PR with a single commit with all the changes so it's easier to read

**Comment by qkdefus** [comment]
using hex the way i did works, kinda tedious converting each hex to int by hand

**Pull request closed without merging**